src/components/userFeedback.js

```diff
@@ -0,0 +1,113 @@
+import '../styles/components/userFeedback.scss'
+
+import React, { useRef, useState } from 'react'
+import HubspotForm from 'react-hubspot-form'
+import Loading from '../components/loading'
+import { FormattedMessage } from 'react-intl'
+import { trackCustomEvent } from 'gatsby-plugin-google-analytics'
+
+const UserFeedback = ({ title, locale }) => {
+  const feedbackBodyRef = useRef(null)
+  const feedbackCloseRef = useRef(null)
+  const [showFollowUp, setShowFollowUp] = useState(false)
+  const [showYesFollowUp, setShowYesFollowUp] = useState(false)
+
+  const setDocHelpful = (docTitle, isHelpful) => {
+    trackCustomEvent({
+      category: isHelpful ? `doc-${locale}-useful` : `doc-${locale}-useless`,
+      action: 'click',
+      label: docTitle,
+      transport: 'beacon',
+    })
+
+    setShowFollowUp(true)
+
+    if (isHelpful) {
+      setShowYesFollowUp(true)
+    }
+  }
+
+  const showThumbs = () => {
+    if (!feedbackCloseRef.current.classList.contains('show-feedback-close')) {
+      feedbackBodyRef.current.classList.add('show-feedback-body')
+      feedbackCloseRef.current.classList.add('show-feedback-close')
+    }
+  }
+
+  const closeFeedback = () => {
+    if (feedbackCloseRef.current.classList.contains('show-feedback-close')) {
+      setShowFollowUp(false)
+      setShowYesFollowUp(false)
+      feedbackBodyRef.current.classList.remove('show-feedback-body')
+      feedbackCloseRef.current.classList.remove('show-feedback-close')
+    }
+  }
+
+  return (
+    <section className="PingCAP-UserFeedback feedback-prompt">
+      <div className="feedback-header">
+        {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions */}
+        <div
+          className="feedback-title"
+          onClick={showThumbs}
+          onKeyDown={showThumbs}
+        >
+          <FormattedMessage id="docHelpful.header" />
+        </div>
+        {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions */}
+        <div
+          className="close-icon"
+          onClick={closeFeedback}
+          onKeyDown={closeFeedback}
+          ref={feedbackCloseRef}
+        >
+          x
+        </div>
+      </div>
+      <div className="feedback-body" ref={feedbackBodyRef}>
```

Comment:
We can use the state to control the `feedback-body` display or not to prevent the **extra** state management.

For example:

```jsx
<div className={`feedback-body${showFollowUp || showYesFollowUp && ' show-feedback-body'}`}>
```

Of course, you can also use a library like [clsx](https://www.npmjs.com/package/clsx) to help write the complicated syntax.